src/base/bittorrent/session.cpp

```diff
@@ -1088,7 +1088,9 @@ void Session::initializeNativeSession()
     // turn them off before `lt::session` ctor to avoid split second effects
     pack.set_bool(lt::settings_pack::enable_upnp, false);
     pack.set_bool(lt::settings_pack::enable_natpmp, false);
+#if (LIBTORRENT_VERSION_NUM < 10204)
     pack.set_bool(lt::settings_pack::upnp_ignore_nonrouters, true);
```

Comment:
Just noticed it today when it broke my compilation. But is your guard correct? IIRC, current libtorrent master still has 1.2.3 version... Even if I'm wrong why just not disable it for entire libtorrent-1.2?

Reply:
>why just not disable it for entire libtorrent-1.2?

Reading the libt PR, if i'm not mistaken it sets this option to true unconditionally, but the docs say that this option is false by default up until now (libt 1.2.3). https://www.libtorrent.org/reference-Settings.html#upnp_ignore_nonrouters

Reply:
Ok. But what about compilation with latest libtorrent RC_1_2? Is it possible with this patch?

Reply:
Basically the timeline looks like this (top to bottom):
```
libtorrent 1.2.3 released
...
libtorrent deprecates upnp_ignore_nonrouters
...
libtorrent 1.2.4 released
```

>Ok. But what about compilation with latest libtorrent RC_1_2? Is it possible with this patch?

Not really, however currently there is not enough information (from c++ code viewpoint) to precisely differentiate at the offending commit.

Reply:
>Not really, however currently there is not enough information (from c++ code viewpoint) to precisely differentiate at the offending commit.

@arvidn?

Reply:
as part of a larger cleanup and fixing of how to deal with multiple listen sockets (most commonly as IPv4 and IPv6) I fixed an issue in UPnP as well. UPnP is now only active for the specific subnets you have listen sockets on, and only care about responses on this subnet (this is currently in `RC_1_2`). In the current stable (1.2.3) UPnP is kind of independent and tries to find gateway devices on all subnets. the option `upnp_ignore_nonrouters` was a way to constrain that a bit to only care about responses from the respective subnet's gateway (if it had one). In all home setups, the gateway will always be the UPNP gateway device anyway. With the new behavior, which is already more constrained, I don't think it's important to constrain it further to only care about (SSDP) responses from the gateway itself.

For this reason I deprecated this option in `RC_1_2`. It may seem a bit drastic for a point release, and maybe it is. However, the ABI is not broken by this. There is still a `deprecated_upnp_ignore_nonrouters` enum with the same value, to ensure this. I would recommend just not offering this setting to users. Even in the old version, it's really a fringe setting.

Reply:
@arvidn, sorry, I still can't understand, how to properly handle it in the code. Current qBt won't compile with RC_1_2 until libtorrent-1.2.4 is released.

Reply:
oh, I see. Because I haven't bumped the version number yet.. hm.

Reply:
as a stop-gap you could use the numeric value (as a plain integer) instead of the enum. The main risk with that is that if I ever remove this option entirely, there would be no build error